scripts/actor/actor-soulbound.js

```diff
@@ -306,11 +306,18 @@ export class SoulboundActor extends WarhammerActor {
     }
 
     speakerData(token) {
-        if (token || this.isToken)
+        if (token)
         {
             return {
-                token : (token?.document || this.token).id,
-                scene : (token?.document || this.token).parent.id
+                token : token.id,
+                scene : token.parent.id
+            }
+        }
+        else if (this.isToken)
+        {
+            return {
+                token : this.token.id,
+                scene : this.token.parent.id
```

Comment:
The only change needed for this to be fixed is the other one in `test.js`, I'm not sure what this is accomplishing?

Reply:
You're right, I went backwards and I think it seemed necessary from that order, but upon doing some testing this can be dropped. Updated.